ansible/library/topo_facts.py

```diff
@@ -18,19 +19,90 @@
       required: True
 '''
 
+def get_vm_type(ngs_type):
+    if ngs_type == "ToRRouter":
+        return "tor"
+    if ngs_type == "LeafRouter":
+        return "leaf"
+    if ngs_type == "SpineRouter":
+        return "spine"
+
```

Comment:
Can you add a comment to explain why the translation/conversion is needed?

Why we cannot use the original ngs_type, but have to use a shortened name? I don't see reference in this change so they must have been used by existing code. If there is a good reason for this have to happen, then it should be documented here as comments.

Reply:
Dropped, by way of re-writing in different fashion.